### ISSUE TITLE
Revisit instant mode UI

### ISSUE THREAD

**Issue by 0xbad1d3a5** [0xbad1d3a5/Kaku#12]
We can probably automatically determine box size in instant mode so users wouldn't have to real with resizing a box. Investigate feasibility.

Similar app that does it: https://www.youtube.com/watch?v=5f5YRc_yRtA&feature=youtu.be

Kaku's interface will probably be slightly different since it still makes a lot of sense to have the InstantKanjiWindows show up for rapid correction, but the idea is the same.

**Comment by SpongebobSquamirez** [comment]
I think dragging being outside the square is maybe okay, but fine-tuning the size of the window requires that you have a good idea of where that window is, and may require adjusting the page you're looking at (e.g. if you're reading an ebook or looking at a webpage), so I actually like that the resize feature lies inside the box, since I know where it is at least and won't accidentally resize when I don't intend to, having to redo the finicky box resizing all over again. If hurigana stripping and character detection (basically...OCR lol) were improved to the level of the [OCR Manga Reader for Android](http://ocrmangareaderforandroid.sourceforge.net/) (source code available), then maybe resizing the window wouldn't be such a delicate task, and this could be introduced. If I heard what kind of use-case benefits from having the resize function outside the box, then maybe I could 納得できるかもしれません。

Just my personal opinion from what I understand right now. Resizing is actually really frustrating, so I'm kind of passionate about this enhancement not being implemented.

**Comment by SpongebobSquamirez** [comment]
I've been using the OCR Manga Reader for Android a lot lately, and it has resizing outside of the box. I was really frustrated by it at first, but I have come to realize that there is also value in seeing the full box so you can see whether you're completely/precisely covering a character you want to OCR. Your finger can get in the way if the place you need to touch is the same place you're trying to precisely align.

I can see value in both, but I think maybe if window-locking were implemented, then having the resize/drag area outside of the square would be good - you'd just have to explain it in the tutorial, because it's definitely not intuitive, and takes a little practice to get used to.

**Comment by 0xbad1d3a5** [comment]
Are you talking about "moving the drag area outside of the box"? I'm actually probably not going to implement that anymore. I found an app that I think does single-line OCR much better and will be trying to have Kaku match that for instant mode (though probably won't start working on it until after the Sakura milestone). Normal mode probably will stay the same since usually the box is big enough those kinds of mistakes aren't made.

I've edited the first post in this issue to show what I mean.

**Comment by SpongebobSquamirez** [comment]
Yeah I was talking about "moving the drag area outside of the box."

Video looks cool. If it proves infeasible though I'm pretty happy the way instant mode is now. It's not like reading/OCRing is a time/workflow bottleneck for me (Ankidroid and J-J dictionaries are my bottlenecks), and I am pleased with the results of instant mode. 

A few questions about the Chinese app:

- Have you testing that app with zoomed-in text? I'm wondering how it performs with various text sizes, large and small. Seems like you need to be able to zoom in/out on the app you're using to match the cursor - but not all apps have that functionality, so it'd be better if it rested with the OCR tool.
- It seems like you need to put it in the bottom left corner of the text you want it to read, but you'd have to have a way to switch it to vertical text mode. I'd think a Chinese app would already have that feature, though, right...?
- How will this handle hurigana? At least with a window the user can make sure they aren't included in the capture. Even if automatic hurigana stripping were implemented, unless a) it's perfect or b) the user has plenty of (perhaps too much...?) freedom in correcting errors (e.g. maybe provide OCR results assuming that the junk above the text was actually hurigana, but also provide results that assume it was a 冠 part of the character), it would be annoying to have to trust the stripping. The OCR Manga Reader is kind of nice with its hurigana detection, but it still lets the user adjust the capture window in case the results aren't perfect, which is a nice balance.

**Comment by 0xbad1d3a5** [comment]
No, I haven't tested their app (not interested in spending $10). There's honestly probably not much difference between Kaku and Hanping in terms of underlying tech, they're just doing slightly more pre-processing. As far as I can tell, they don't have a vertical mode.

I'm not expecting stripping furigana to be an issue. I've honestly had pretty decent results when I was testing some of my [experimental code](https://github.com/0xbad1d3a5/Kaku/blob/652d0d622c82baed3d0517634a49a74d9105f776/app/src/main/java/ca/fuwafuwa/kaku/Windows/CaptureWindow.java#L89).

Biggest downside is probably needing to scrap 70% of [CaptureWindow](https://github.com/0xbad1d3a5/Kaku/blob/652d0d622c82baed3d0517634a49a74d9105f776/app/src/main/java/ca/fuwafuwa/kaku/Windows/CaptureWindow.java#L42) which is a pain.
